tests/core/test_geometric_sources.py

```diff
@@ -74,3 +74,158 @@ def test_translate_direction_collinear(is_negative, delta, bunny):
+        # We also use a symmetric text string since the oriented mesh's
+        # bounding box center and/or the mean of its points will otherwise
+        # vary and is challenging to test
+        new_normal = np.array((1, -2, 3))
+        src = pv.Text3DSource(string='I', center=center, normal=new_normal, depth=0)
+        out = src.output
+        actual_normal = np.mean(out.cell_normals, axis=0)
+        expected_normal = new_normal / np.linalg.norm(new_normal)
+        assert np.allclose(actual_normal, expected_normal, atol=1e-4)
+
+        points_center = np.mean(out.points, axis=0)
+        assert np.allclose(points_center, center, atol=1e-4)
+
+
+@pytest.mark.parametrize(
+    'kwarg',
+    [
+        ('string', 'TEXT'),
+        ('center', (1, 2, 3)),
+        ('normal', (4, 5, 6)),
+        ('height', 2),
+        ('width', 3),
+        ('depth', 4),
+    ],
+)
+def test_text3d_source_modified(kwarg):
+    # Set up test param
+    name, value = kwarg
+    test_kwargs = {name: value}
+
+    # Make sure string is always set to ensure non-empty output
+    string_val = 'TEXT'
+    test_kwargs.setdefault('string', string_val)
+
+    # Test init modifies source but does not update output
+    src = pv.Text3DSource(**test_kwargs)
+    assert src._modified
+    assert src._output.n_points == 0
+
+    # Store initial object references
+    out1 = src._output
+    out1_points = out1.GetPoints()
+
+    # Test getting output triggers an update and resets modified flag
+    out2 = src.output
+    out2_points = out2.GetPoints()
+    assert not src._modified
+    assert len(out1.split_bodies()) == len(string_val)
+    assert out2 is out1  # same output obj
+    assert out2_points is not out1_points  # new points array
+
+    # Test setting the same value does not modify obj
+    setattr(src, name, value)
+    assert not src._modified
+    assert src._output is out1
+    assert src._output.GetPoints() is out2_points
+    assert len(src._output.split_bodies()) == len(string_val)
+
+    # Test calling update has no effect on output when modified flag is not set
+    src.update()
+    assert not src._modified
+    assert src._output is out1
+    assert src._output.GetPoints() is out2_points
+    assert len(src._output.split_bodies()) == len(string_val)
+
+    # Test setting to a new value sets flag but does not modify the output
+    if name == 'string':
+        new_value = value + value
+    else:
+        new_value = np.array(value) * 2
+    assert not np.array_equal(new_value, value)
+
+    setattr(src, name, new_value)
+    assert src._modified
+    assert src._output is out1
+    assert src._output.GetPoints() is out2_points
+    assert len(src._output.split_bodies()) == len(string_val)
+
+    # Test calling update modifies the output
+    src.update()
+    assert not src._modified
+    assert src._output is out1
+    assert src._output.GetPoints() is not out2_points
+    assert len(src._output.split_bodies()) == len(src.string)
```

Comment:
In my mind this test is perfect. However, the long function lines make it difficult for the reviewer to understand the scope of the variables. I am also concerned that the previous test might affect the test behind it. Could you please make the Text3DSource definition a fixture and split the test function into separate sections?